hazelcast/src/test/java/com/hazelcast/spi/impl/operationservice/impl/InvocationFuture_AndThenTest.java

```diff
@@ -59,6 +63,31 @@ public void whenNullCallback() {
         future.andThen(null);
     }
 
+    @Test(expected = HazelcastInstanceNotActiveException.class)
+    public void whenNodeIsShutdown() throws Throwable {
+        Address address = getAddress(local);
+        local.shutdown();
+
+        DummyOperation op = new DummyOperation(null);
+        InternalCompletableFuture<Object> future = operationService.invokeOnTarget(null, op, address);
+        CountDownLatch latch = new CountDownLatch(1);
+        AtomicReference<Throwable> reference = new AtomicReference<>();
+        future.andThen(new ExecutionCallback<Object>() {
+            @Override
+            public void onResponse(Object response) {
+
+            }
+
+            @Override
+            public void onFailure(Throwable t) {
+                reference.set(t);
+                latch.countDown();
+            }
+        });
+        assertOpenEventually(latch);
+        throw reference.get();
```

Comment:
Isn't it better to inspect exception directly here, such as; `assertInstanceOf(...)`, instead of throwing it?  With this style, `HazelcastInstanceNotActiveException ` might be thrown from another place in the test, yet test still passes.